design-toolkit/components/src/components/table/table-body.tsx

```diff
@@ -10,17 +10,52 @@
  * governing permissions and limitations under the License.
  */
 
+import type { Row, RowData } from '@tanstack/react-table';
+import { useContext } from 'react';
+import { TableContext } from '.';
 import { tableBodyStyles } from './styles';
+import { TableRow } from './table-row';
 import type { TableBodyProps } from './types';
 
 export function TableBody({ className, ref, ...props }: TableBodyProps) {
+  const { getTopRows, getCenterRows, getBottomRows } = useContext(TableContext);
+
+  if (!getCenterRows.length) {
+    return <tbody>{props.children}</tbody>;
+  }
+
   return (
     <tbody
       ref={ref}
       className={tableBodyStyles({
         className,
       })}
       {...props}
-    />
+    >
+      {getTopRows().map((row: Row<RowData>) => (
+        <TableRow
+          key={row.id}
+          row={row}
+          {...(row.getIsSelected() ? { 'data-selected': '' } : {})}
+          data-pinned={row.getIsPinned()}
+        />
+      ))}
+      {getCenterRows().map((row: Row<RowData>) => (
+        <TableRow
+          key={row.id}
+          row={row}
+          {...(row.getIsSelected() ? { 'data-selected': '' } : {})}
+          data-pinned={row.getIsPinned()}
+        />
+      ))}
+      {getBottomRows().map((row: Row<RowData>) => (
```

Comment:
Considering all three of these row maps are identical, couldn't we just concat the arrays and have a single map?